Merge node-private-preview

jar-stripe opened the pull request:
### Why?
<!-- Describe why this change is being made.  Briefly include history and context, high-level what this PR does, and what the world looks like afterward. -->

### What?
* ran `just merge node-private-preview` from codegen folder
<!--
List out the key changes made in this PR, e.g.
- implements the antimatter particle trace in the nitronium microfilament drive
- updated tests -->

### See Also
<!-- Include any links or additional information that help explain this change. -->

mbroshi-stripe commented:
@jar-stripe I tried updating the base to `private-preview` but GH tells me there are no commits.

jar-stripe commented:
> @jar-stripe I tried updating the base to `private-preview` but GH tells me there are no commits.

ah i forgot to push.  fixed it and updated the base.  thanks!
